src/ast/irbuilderbpf.cpp

```diff
@@ -1491,9 +1492,8 @@ Value *IRBuilderBPF::CreateUSDTReadArgument(Value *ctx,
   }
 
   if (usdt == nullptr) {
-    builtin.addError() << "failed to initialize usdt context for probe "
-                       << attach_point->target;
-    exit(-1);
+    throw util::FatalUserException(
```

Comment:
why the change to a throw here?

Reply:
(I wrote reasons in the commit message :grinning: )

Previously exit() terminates the process immediately, leaving no chance
to output error buffer. This patch instead raises the FatalUserException
with useful information to let users known arg12+ is yet supported by
libbpf.

Reply:
Ah my apologies, I missed the `exit(-1)`. Just curious, what happens if we just return and not throw or exit? I see that we're adding an error to the builtin - hopefully the passmanger should recognize this and terminate the execution anyway with the correct messaging.

Reply:
> what happens if we just return

I tried before, bpftrace crashed with segfault.

Reply:
Ah ok, no worries. Throwing is fine then. I'll take a look at it later.